src/components/engagement/engagement.service.ts

```diff
@@ -982,12 +945,36 @@ export class EngagementService {
         canRead: canReadEndDate,
         canEdit: false,
       },
+      methodologies: {
+        ...securedProperties.methodologies,
+        value: securedProperties.methodologies.value ?? [],
+      },
+      ceremony: {
+        ...securedProperties.ceremony,
+        value: result.ceremonyId,
+      },
+      language: {
+        ...securedProperties.language,
+        value: result.languageId,
+      },
+      intern: {
+        ...securedProperties.intern,
+        value: result.internId,
+      },
+      countryOfOrigin: {
+        ...securedProperties.countryOfOrigin,
+        value: result.countryOfOriginId,
+      },
+      mentor: {
+        ...securedProperties.mentor,
+        value: result.mentorId,
+      },
     };
 
-    if (result.__typename === 'LanguageEngagement') {
-      return (response as unknown) as LanguageEngagement;
-    } else if (result.__typename === 'InternshipEngagement') {
-      return (response as unknown) as InternshipEngagement;
+    if (engagement.__typename === 'LanguageEngagement') {
+      return (engagement as unknown) as LanguageEngagement;
+    } else if (engagement.__typename === 'InternshipEngagement') {
+      return (engagement as unknown) as InternshipEngagement;
     } else {
       throw new NotFoundException('could not find Engagement');
     }
```

Comment:
Is this still needed?

Reply:
Michael said yes (a few refactors ago) but I don't know.  I'll defer to your expertise

Reply:
Typecasting is just to make TS happy. If it can be removed, it should be. As long as the return type of the function doesn't change.

Reply:
I think this is just for TS.

Reply:
removed this typecasting